odl/discr/lp_discr.py

```diff
@@ -446,8 +528,16 @@ def __setitem__(self, indices, values):
             shape is allowed as ``values``.
         """
         if values in self.space:
-            self.ntuple.__setitem__(indices, values.ntuple)
+            # For RawDiscretizationVector of the same type, use ntuple directly
+            return self.ntuple.__setitem__(indices, values.ntuple)
+        elif (isinstance(self.space.dspace, Ntuples) and
+              equiv_views(self.asarray(), values) and
```

Comment:
This cannot be merged. It makes some operations exponentially slower, e.g.

```
>>> X = odl.CudaRn(10**7)
>>> x = X.element()
>>> x[0] = 2
```

currently this is almost instant, with this change this would be an O(n) operation.